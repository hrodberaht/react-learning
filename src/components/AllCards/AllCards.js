import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Card from './Card/Card';
import { deleteCard } from '../../actions';

const mapStateToProps = state => ({
  cards: state.cards,
  filterText: state.filterText,
});

const mapDispatchToProps = dispatch => ({
  handleClick: id => dispatch(deleteCard(id)),
});
class AllCards extends Component {
  render() {
    const { cards, handleClick, filterText } = this.props;
    let list = cards;
    if (filterText) {
      list = cards.filter(card => card.name.toLowerCase().includes(filterText));
    }

    return (
      <div>
        {list.map(card => (
          <Card
            key={card.id}
            card={card}
            handleClick={handleClick}
          />
        ))}
      </div>
    );
  }
}

AllCards.propTypes = {
  cards: PropTypes.arrayOf(PropTypes.object).isRequired,
  handleClick: PropTypes.func.isRequired,
  filterText: PropTypes.string.isRequired,
};
export { AllCards };
export const ConnectedAllCards = connect(mapStateToProps, mapDispatchToProps)(AllCards);
