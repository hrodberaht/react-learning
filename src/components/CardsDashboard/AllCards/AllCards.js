import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Card from './Card/Card';
import { deleteCard } from '../../../actions';

export class AllCards extends Component {
  list = () => {
    const { cards, filterText } = this.props;
    if (filterText) {
      const regexp = new RegExp(filterText, 'i');
      return cards.filter(card => regexp.test(card.name));
    }
    return cards;
  };

  render() {
    const { handleClick } = this.props;
    return (
      <div className="cards-list">
        {this.list().map(card => (
          <Card
            key={card.id}
            card={card}
            handleClick={handleClick}
          />
        ))}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  cards: state.cards,
  filterText: state.filterText,
});

export default connect(
  mapStateToProps,
  {
    handleClick: deleteCard,
  },
)(AllCards);

AllCards.propTypes = {
  cards: PropTypes.arrayOf(PropTypes.object).isRequired,
  handleClick: PropTypes.func.isRequired,

  filterText: PropTypes.string.isRequired,
};
