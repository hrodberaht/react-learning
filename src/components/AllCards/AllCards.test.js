import React from 'react';
import { shallow } from 'enzyme';
import { AllCards } from './AllCards';

describe('<AllCards />', () => {
  let wrapper;
  const props = {
    cards: [
      {
        id: 1,
        name: 'John',
        email: 'john@gmail.com',
      },
      {
        id: 2,
        name: 'Rob',
        email: 'rob@gmail.com',
      },
    ],
    handleClick: jest.fn(),
    filterText: '',
  };

  beforeEach(() => {
    wrapper = shallow(<AllCards {...props} />);
  });

  it('should render component witout crash', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render two cards when filtertext is empty string', () => {
    const cards = wrapper.find('Card');

    expect(cards.length).toBe(2);
  });

  it('should render cards when filterText is not empty', () => {
    wrapper.setProps({ filterText: 'John' });
    const cards = wrapper.find('Card');

    expect(cards.length).toBe(1);
  });
});
