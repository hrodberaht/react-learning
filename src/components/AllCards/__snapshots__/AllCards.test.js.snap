// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AllCards /> should render component witout crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AllCards
    cards={
        Array [
            Object {
              "email": "john@gmail.com",
              "id": 1,
              "name": "John",
            },
            Object {
              "email": "rob@gmail.com",
              "id": 2,
              "name": "Rob",
            },
          ]
    }
    filterText=""
    handleClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Card
          card={
                    Object {
                              "email": "john@gmail.com",
                              "id": 1,
                              "name": "John",
                            }
          }
          handleClick={[Function]}
/>,
        <Card
          card={
                    Object {
                              "email": "rob@gmail.com",
                              "id": 2,
                              "name": "Rob",
                            }
          }
          handleClick={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "card": Object {
            "email": "john@gmail.com",
            "id": 1,
            "name": "John",
          },
          "handleClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "card": Object {
            "email": "rob@gmail.com",
            "id": 2,
            "name": "Rob",
          },
          "handleClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Card
            card={
                        Object {
                                    "email": "john@gmail.com",
                                    "id": 1,
                                    "name": "John",
                                  }
            }
            handleClick={[Function]}
/>,
          <Card
            card={
                        Object {
                                    "email": "rob@gmail.com",
                                    "id": 2,
                                    "name": "Rob",
                                  }
            }
            handleClick={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "card": Object {
              "email": "john@gmail.com",
              "id": 1,
              "name": "John",
            },
            "handleClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "card": Object {
              "email": "rob@gmail.com",
              "id": 2,
              "name": "Rob",
            },
            "handleClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
